libs/libcommon/src/libcommon/processing_graph.py

```diff
@@ -39,6 +39,24 @@ def guard_int(x: Any) -> int:
     raise ValueError(f"Invalid int: {x}")
 
 
+def check_one_of_parents_is_same_or_higher_level(processing_graph: ProcessingGraph) -> None:
+    for step in processing_graph.get_topologically_ordered_processing_steps():
+        parents = processing_graph.get_parents(step.name)
+        if len(parents) > 0:
```

Comment:
Is it worth validating that steps without parents are part of _first_processing_steps?

Reply:
well, it does not cost much, I'll add it